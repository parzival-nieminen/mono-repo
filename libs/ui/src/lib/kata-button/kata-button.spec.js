import {fireEvent, render, screen} from '@testing-library/react';
import KataButton from './kata-button';
import renderer from 'react-test-renderer';

describe('KataButton', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<KataButton/>);
    expect(baseElement).toBeTruthy();
  });
  it('is disabled', () => {
    const tree = renderer.create(<KataButton disabled text={"My Button"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has no label as text', () => {
    const tree = renderer.create(<KataButton/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with subtext no data selected', () => {
    const tree = renderer.create(<KataButton text={"Save"}>no data selected</KataButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('onClick called 1 and Save arg is included', () => {
    const func = jest.fn()
    render(<KataButton text={"Save"} onClick={func}>no data selected</KataButton>)

    fireEvent.click(screen.getByRole("button"))

    expect(func).toHaveBeenCalledWith(expect.anything(), "Button 'Save' is called")
    expect(func).toHaveBeenCalledTimes(1)
  });
});
