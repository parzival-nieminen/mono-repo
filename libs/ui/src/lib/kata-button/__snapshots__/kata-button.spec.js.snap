// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KataButton has no label as text 1`] = `
Array [
  <div
    className="KataButton"
    onClick={[Function]}
    role="button"
  >
    no label
  </div>,
  <p />,
]
`;

exports[`KataButton is disabled 1`] = `
Array [
  <div
    className="Disabled"
    onClick={[Function]}
    role="button"
  >
    My Button
  </div>,
  <p />,
]
`;

exports[`KataButton with subtext no data selected 1`] = `
Array [
  <div
    className="KataButton"
    onClick={[Function]}
    role="button"
  >
    Save
  </div>,
  <p>
    no data selected
  </p>,
]
`;
